### DISCUSSION TITLE
unable to build marp-vscode

### DISCUSSION THREAD

**Discussion by kongdd** [marp-team/marp#387] [Q&A] [answered]
I am trying to build marp-vscode on my local PC. However, failed.

```bash
git clone https://github.com/marp-team/marp-vscode
cd marp-vscode
npm install
vsce publish
```
But I got the following error:

```typescript
Executing prepublish script 'npm run vscode:prepublish'...

> marp-vscode@2.3.0 vscode:prepublish
> run-s preversion build


> marp-vscode@2.3.0 preversion
> run-p check:* lint:* test:*:coverage


> marp-vscode@2.3.0 check:audit
> npm audit


> marp-vscode@2.3.0 lint:js
> eslint --ext .js,.ts --report-unused-disable-directives --cache .


> marp-vscode@2.3.0 check:format
> npm run -s format -- -c


> marp-vscode@2.3.0 test:unit:coverage
> jest --coverage


> marp-vscode@2.3.0 check:ts
> tsc --noEmit


> marp-vscode@2.3.0 lint:css
> stylelint "./*.{css,scss}"


 RUNS  src/extension.test.ts
 RUNS  src/commands/export.test.ts
 RUNS  src/language/completions.test.ts
 RUNS  src/extension.test.ts
 RUNS  src/commands/export.test.ts
 RUNS  src/language/completions.test.ts
 RUNS  src/language/parser.test.ts
 RUNS  src/diagnostics/define-math-global-directive.test.ts
 RUNS  src/diagnostics/deprecated-dollar-prefix.test.ts
 RUNS  src/diagnostics/deprecated-color-setting-shorthand.test.ts
 RUNS  src/preview.test.ts
 RUNS  src/option.test.ts
 RUNS  src/marp-cli.test.ts
 RUNS  src/language/index.test.ts
 RUNS  src/extension.test.ts
 RUNS  src/commands/export.test.ts
 RUNS  src/language/completions.test.ts
 RUNS  src/language/parser.test.ts
 RUNS  src/diagnostics/define-math-global-directive.test.ts
 RUNS  src/diagnostics/deprecated-dollar-prefix.test.ts
 RUNS  src/diagnostics/deprecated-color-setting-shorthand.test.ts
 RUNS  src/preview.test.ts
 RUNS  src/option.test.ts
 RUNS  src/marp-cli.test.ts
 RUNS  src/language/index.test.ts
 RUNS  src/commands/toggle-marp-feature.test.ts
 RUNS  src/diagnostics/index.test.ts
 RUNS  src/diagnostics/unknown-size.test.ts
 RUNS  src/diagnostics/unknown-theme.test.ts

Test Suites: 0 of 23 total
Tests:       0 total
Snapshots:   0 total
[warn] CHANGELOG.md
ERROR: "check:audit" exited with 1.
ERROR: "preversion" exited with 1.
 ERROR  npm failed with exit code 1
```

**Comment by yhatt** [comment, marked as answer]
`vsce publish` is for updating [our extension on Visual Studio marketplace](https://marketplace.visualstudio.com/items?itemName=marp-team.marp-vscode). Running publish command is allowed only to a project maintainer who have privileges to bump version and release.

Recommend to use [VS Code debugger for build and development](https://github.com/marp-team/marp-vscode/blob/main/.github/CONTRIBUTING.md#development), but you can use `npm run build` if you want just to build (transpile TS to JS).

**Comment by kongdd** [comment]
I have made some changes for marp-vscode. I just need to build the VSIX for self use.
Could you give some guidance how to make it?

**Reply by yhatt** [reply]
Did you read [Marp for VS Code contributing guideline](https://github.com/marp-team/marp-vscode/blob/main/.github/CONTRIBUTING.md#package-vsix:~:text=supporting%20E2E%20tests.-,Package%20VSIX,-npm%20run%20package)?

**Comment by kongdd** [comment]
Thanks for your reminder. My problem has been solved by changing:
~~`"vscode:prepublish": "run-s preversion build"`~~ into 
`"vscode:prepublish": "run-s build"`

**Reply by yhatt** [reply]
It means to skip every checks for production: linting, testing, vulnerability check, etc. Marp team assumes no responsibility to every happens by your custom build 😛
